diff --git a/desktop/gimp.desktop.in.in b/desktop/gimp.desktop.in.in
index 776e4ee..0e92763 100644
--- a/desktop/gimp.desktop.in.in
+++ b/desktop/gimp.desktop.in.in
@@ -4,6 +4,8 @@ Type=Application
 _Name=GNU Image Manipulation Program
 _GenericName=Image Editor
 _Comment=Create images and edit photographs
+# Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
+_Keywords=GIMP;graphic;design;illustration;painting;
 Exec=@GIMP_COMMAND@ %U
 TryExec=gimp-@GIMP_APP_VERSION@
 Icon=gimp
