diff -up gegl-2/gegl/gegl-enums.h.mkenums gegl-2/gegl/gegl-enums.h
--- gegl-2/gegl/gegl-enums.h.mkenums	2018-04-06 08:52:15.075045457 +0200
+++ gegl-2/gegl/gegl-enums.h	2018-04-06 08:56:49.064313865 +0200
@@ -32,10 +32,10 @@
 G_BEGIN_DECLS
 
 typedef enum {
-  GEGL_SAMPLER_NEAREST = 0,   /*< desc="nearest"      >*/
-  GEGL_SAMPLER_LINEAR,        /*< desc="linear"       >*/
-  GEGL_SAMPLER_CUBIC,         /*< desc="cubic"        >*/
-  GEGL_SAMPLER_LOHALO         /*< desc="lohalo"       >*/
+  GEGL_SAMPLER_NEAREST = 0,   /*< desc="nearest" , nick=nearest >*/
+  GEGL_SAMPLER_LINEAR,        /*< desc="linear" , nick=linear >*/
+  GEGL_SAMPLER_CUBIC,         /*< desc="cubic"  , nick=cubic >*/
+  GEGL_SAMPLER_LOHALO         /*< desc="lohalo" , nick=lohalo >*/
 } GeglSamplerType;
 GType gegl_sampler_type_get_type   (void) G_GNUC_CONST;
 #define GEGL_TYPE_SAMPLER_TYPE (gegl_sampler_type_get_type())
@@ -59,13 +59,13 @@ GType gegl_ripple_wave_type_get_type   (
 
 typedef enum
 {
-  GEGL_WARP_BEHAVIOR_MOVE,      /*< desc="Move pixels"              >*/
-  GEGL_WARP_BEHAVIOR_GROW,      /*< desc="Grow area"                >*/
-  GEGL_WARP_BEHAVIOR_SHRINK,    /*< desc="Shrink area"              >*/
-  GEGL_WARP_BEHAVIOR_SWIRL_CW,  /*< desc="Swirl clockwise"          >*/
-  GEGL_WARP_BEHAVIOR_SWIRL_CCW, /*< desc="Swirl counter-clockwise"  >*/
-  GEGL_WARP_BEHAVIOR_ERASE,     /*< desc="Erase warping"            >*/
-  GEGL_WARP_BEHAVIOR_SMOOTH     /*< desc="Smooth warping"           >*/
+  GEGL_WARP_BEHAVIOR_MOVE,      /*< desc="Move pixels"     ,  nick=move       >*/
+  GEGL_WARP_BEHAVIOR_GROW,      /*< desc="Grow area"       ,  nick=grow       >*/
+  GEGL_WARP_BEHAVIOR_SHRINK,    /*< desc="Shrink area"     ,  nick=shrink       >*/
+  GEGL_WARP_BEHAVIOR_SWIRL_CW,  /*< desc="Swirl clockwise" ,  nick=swirl-cw       >*/
+  GEGL_WARP_BEHAVIOR_SWIRL_CCW, /*< desc="Swirl counter-clockwise" , nick=swirl-ccw >*/
+  GEGL_WARP_BEHAVIOR_ERASE,     /*< desc="Erase warping"      ,   nick=erase   >*/
+  GEGL_WARP_BEHAVIOR_SMOOTH     /*< desc="Smooth warping"     ,   nick=smooth   >*/
 } GeglWarpBehavior;
 GType gegl_warp_behavior_get_type (void) G_GNUC_CONST;
 #define GEGL_TYPE_WARP_BEHAVIOR (gegl_warp_behavior_get_type ())
